### DISCUSSION TITLE
Unable to download dependency from GitHub Package Registry

### DISCUSSION THREAD

**Discussion by ADarkDividedGem** [community/community#25517] [Packages]
<p>I have successfully deployed my package to the GitHub Package Registry:</p>
<p><a href="https://github.com/ADarkDividedGem/Google-Play-API/packages/38461?version=1.1" title="com.akdeniz.googleplaycrawler 1.1" rel="nofollow noopener">com.akdeniz.googleplaycrawler 1.1</a> </p>
<p>However, when I add the following dependency into my POM.xml file, Maven is unable to download it:</p>
<pre><code>&lt;dependency&gt;
	&lt;groupId&gt;com.akdenizcom.github.adarkdividedgem&lt;/groupId&gt;
	&lt;artifactId&gt;.googleplaycrawler&lt;/artifactId&gt;
	&lt;version&gt;1.1&lt;/version&gt;
&lt;/dependency&gt;
</code></pre>
<p>A simple “mvn install” fails with the following warning:</p>
<pre><code>Downloading: https://maven.pkg.github.com/ADarkDividedGem/com/github/adarkdividedgem/com.akdeniz.googleplaycrawler/1.1/com.akdeniz.googleplaycrawler-1.1.pom
[WARNING] The POM for com.github.adarkdividedgem:com.akdeniz.googleplaycrawler:jar:1.1 is missing, no dependency information available
</code></pre>
<p>Checking my .m2 repository folder confirms that neither the POM or JAR file was downloaded.</p>
<p>Any idea what I am doing wrong because I am using the same Maven settings.xml file that I used to upload the package which includes all the server and repositry information described in the documentation.</p>

**Comment by ADarkDividedGem** [comment, marked as answer]
<p>I managed to get it working by changing the dependancy to this:</p>
<pre><code>&lt;dependency&gt;
	&lt;groupId&gt;com.akdeniz&lt;/groupId&gt;
	&lt;artifactId&gt;googleplaycrawler&lt;/artifactId&gt;
	&lt;version&gt;1.1&lt;/version&gt;
&lt;/dependency&gt;
</code></pre>
<p>This is different from the GroupID and ArtifactID given by the GitHub Package Registry web page.</p>
<p>Then include the repository name in the repository tag:</p>
<pre><code>&lt;profile&gt;
      &lt;id&gt;github&lt;/id&gt;
      &lt;repositories&gt;
        &lt;repository&gt;
          &lt;id&gt;central&lt;/id&gt;
          &lt;url&gt;https://repo1.maven.org/maven2&lt;/url&gt;
          &lt;releases&gt;&lt;enabled&gt;true&lt;/enabled&gt;&lt;/releases&gt;
          &lt;snapshots&gt;&lt;enabled&gt;true&lt;/enabled&gt;&lt;/snapshots&gt;
        &lt;/repository&gt;
        &lt;repository&gt;
          &lt;id&gt;github&lt;/id&gt;
          &lt;name&gt;GitHub OWNER Apache Maven Packages&lt;/name&gt;
          &lt;url&gt;https://maven.pkg.github.com/adarkdividedgem/Google-Play-API&lt;/url&gt;
		  &lt;releases&gt;&lt;enabled&gt;true&lt;/enabled&gt;&lt;/releases&gt;
          &lt;snapshots&gt;&lt;enabled&gt;true&lt;/enabled&gt;&lt;/snapshots&gt;
        &lt;/repository&gt;
      &lt;/repositories&gt;
    &lt;/profile&gt;
</code></pre>
<p>In the documentation only the OWNER section was mentioned.</p>

**Comment by TobseF** [comment]
<p>I’m also trying to add my deployed maven artfact as dependy. I also set up the GitHub CI which deployed it to the  Package Registry here:</p>
<p><a href="https://github.com/TobseF/HelloMaven/packages/39042" rel="nofollow noopener">https://github.com/TobseF/HelloMaven/packages/39042</a></p>
<p>But how to add it as dependy?</p>
<p><strong>pom.xml</strong></p>
<pre><code>\&lt;repositories\&gt;  
 \&lt;repository\&gt;  
 \&lt;id\&gt;github\&lt;/id\&gt;  
 \&lt;name\&gt;GitHub TobseF Apache Maven Packages\&lt;/name\&gt;  
 \&lt;url\&gt;https://maven.pkg.github.com/TobseF/HelloMaven\&lt;/url\&gt;  
 \&lt;releases\&gt;\&lt;enabled\&gt;true\&lt;/enabled\&gt;\&lt;/releases\&gt;  
 \&lt;snapshots\&gt;\&lt;enabled\&gt;true\&lt;/enabled\&gt;\&lt;/snapshots\&gt;  
 \&lt;/repository\&gt;  
\&lt;/repositories\&gt;

\&lt;dependency\&gt;  
 \&lt;groupId\&gt;github.tobsef\&lt;/groupId\&gt;  
 \&lt;artifactId\&gt;hello-maven\&lt;/artifactId\&gt;  
 \&lt;version\&gt;1.2.1\&lt;/version\&gt;  
\&lt;/dependency\&gt;
</code></pre>
<p>Which did I missed? Do I also need to setup a <strong>settings.xml</strong> with credentials?</p>
<p>I also tried it in the project and globally in this way:</p>
<pre><code>&lt;activeProfiles&gt;
        &lt;activeProfile&gt;github&lt;/activeProfile&gt;
    &lt;/activeProfiles&gt;

    &lt;profiles&gt;
        &lt;profile&gt;
            &lt;id&gt;github&lt;/id&gt;
            &lt;repositories&gt;
                &lt;repository&gt;
                    &lt;id&gt;github&lt;/id&gt;
                    &lt;name&gt;GitHub TobseF Apache Maven Packages&lt;/name&gt;
                    &lt;url&gt;https://maven.pkg.github.com/TobseF&lt;/url&gt;
                &lt;/repository&gt;
            &lt;/repositories&gt;
        &lt;/profile&gt;
    &lt;/profiles&gt;

    &lt;servers&gt;
        &lt;server&gt;
            &lt;id&gt;github&lt;/id&gt;
            &lt;username&gt;TobseF&lt;/username&gt;
            &lt;password&gt;123-my-auth-toke-123&lt;/password&gt;
        &lt;/server&gt;
    &lt;/servers&gt;
</code></pre>
<p>But I had no succces. Did you set up the authetification an how?</p>
<p>Can you add my sample lib as dependency?</p>
<p>Best wishes,</p>
<p>Tobse</p>

**Comment by TobseF** [comment]
<p>OK, I aso got it working, by removing the <strong>.m2/settings.xml</strong> in the project and suing the global <strong>username/.m2/settings.xml.</strong></p>
<p>There I changed repository url from </p>
<pre><code>https://maven.pkg.github.com/TobseF/
</code></pre>
<p>to</p>
<pre><code>https://maven.pkg.github.com/TobseF/HelloMaven
</code></pre>
<p>so I have:</p>
<pre><code>&lt;profiles&gt;
    &lt;profile&gt;
      &lt;id&gt;github&lt;/id&gt;
      &lt;repositories&gt;
			&lt;repository&gt;
				&lt;id&gt;github&lt;/id&gt;
				&lt;name&gt;GitHub OWNER Apache Maven Packages&lt;/name&gt;
				&lt;url&gt;https://maven.pkg.github.com/TobseF/HelloMaven&lt;/url&gt;
				&lt;releases&gt;&lt;enabled&gt;true&lt;/enabled&gt;&lt;/releases&gt;
				&lt;snapshots&gt;&lt;enabled&gt;true&lt;/enabled&gt;&lt;/snapshots&gt;
			&lt;/repository&gt;
      &lt;/repositories&gt;
    &lt;/profile&gt;
  &lt;/profiles&gt;

  &lt;servers&gt;
    &lt;server&gt;
      &lt;id&gt;github&lt;/id&gt;
      &lt;username&gt;TobseF&lt;/username&gt;
      &lt;password&gt;123-my-toke-123&lt;/password&gt;
    &lt;/server&gt;
  &lt;/servers&gt;
</code></pre>
<p>So to add a dependency, do I really need to mention every GitHub Registry depency in my global <strong>settings.xml</strong>?</p>
<p>At the end of the day I have to list there hundreds.</p>
<p>Or is it enough to set up my token for the <strong>id</strong> named <strong>github</strong> and hope everybody is using this name in their  <strong>pom.xml</strong> -good luck?</p>

**Comment by Blackdread** [comment]
<p>Still not working</p>

**Comment by seelenfaenger** [comment]
<p>It appears the github page showing how to set up the dependency in a maven pom.xml file simply lies :wink:<br>
Use the groupId and the artifactId from your project and it should work. At least its what I encountered:</p>
<ul>
<li>project which was published had the groupId set to a.b</li>
<li>project which was published had the artifactId set to c.d</li>
<li>after publishing the package page from GitHub lied to me that the groupId is a.b.c and the artifact id is d</li>
<li>using the groupId a.b and the artifactId c.d in the dependency of another project downloaded the jar correctly</li>
</ul>
<p>Hope it helps</p>
